crates/matrix-sdk-base/src/client.rs

```diff
@@ -496,13 +498,18 @@ impl BaseClient {
                                 SyncMessageLikeEvent::Original(original_event),
                             ) => match &original_event.content.msgtype {
                                 MessageType::VerificationRequest(_) => {
-                                    Box::pin(self.handle_verification_event(e, room.room_id()))
-                                        .await?;
+                                    if !ignore_verification_requests {
+                                        Box::pin(self.handle_verification_event(e, room.room_id()))
```

Comment:
Shouldn't we use this `ignore_verification_requests` inside the `handle_verification_event` method instead of here?

Reply:
To me it feels a bit weird that a method called `handle_verification_event` gets a parameter telling it if it should ignore the event, but I can move it there if you'd rather centralise it. However, this still needs to get passed as a parameter since `BaseClient` shouldn't hold this value.

Reply:
What's the reasoning for not putting the parameter into the `BaseClient`?

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/4705#discussion_r1967996296